rllib/models/torch/visionnet.py

```diff
@@ -131,7 +131,18 @@ def __init__(
                 padding, _ = same_padding(in_size, [1, 1], [1, 1])
                 if post_fcnet_hiddens:
                     layers.append(nn.Flatten())
-                    in_size = out_channels
+                    # If we have a post-fc-stack, we can allow arbitrary shapes for the final Conv2D layer.
+                    # So we use a dummy sample to determine the size of the Conv2D output (= post-fc-stack input)
+                    self._convs = nn.Sequential(*layers)
+                    # Create a B=1 dummy sample and push it through out conv-net.
+                    dummy_in = (
+                        torch.from_numpy(self.obs_space.sample())
+                        .permute(2, 0, 1)
```

Comment:
I'm not sure if I understand this permute op. Could you comment here?

Reply:
Good question, I took this code from a little further below, now [line 187](https://github.com/ray-project/ray/pull/22137/files#diff-1dc353522cd53caf7f37edc09d36d7b50c692fd24c628eb94cafd6f0baeb2c6cR187). I assume it's that torch expects channel first, whereas the environment usually has channel last (e.g. 84x84x4 observations in Atari), I admit I didn't look into it in detail though.